web/settings/settings.go

```diff
@@ -123,6 +123,24 @@ func (h *HTTPHandler) postEmail(c echo.Context) error {
 	}
 }
 
+func (h *HTTPHandler) getEmailConfirmation(c echo.Context) error {
+	token := c.QueryParam("token")
+	inst := middlewares.GetInstance(c)
+
+	err := h.svc.ConfirmEmailUpdate(inst, token)
+	switch {
+	case err == nil:
+		// Redirect to the setting page
+		c.Redirect(http.StatusTemporaryRedirect, inst.SubDomain("settings").String())
+	case errors.Is(err, csettings.ErrNoPendingEmail), errors.Is(err, instance.ErrInvalidToken):
+		return jsonapi.BadRequest(errors.New("invalid token"))
```

Comment:
We should show a proper error page here. Something like https://github.com/cozy/cozy-stack/blob/master/web/auth/magic_link.go#L33-L45

Reply:
Yep, this would be great. I will add this